src/conf_mode/syslog.py

```diff
@@ -0,0 +1,267 @@
+        config_data['files'][filename]['selectors']  = generate_selectors(c, 'file ' + filename + ' facility')
+      if c.exists('file ' + filename + ' archive size'):
+       config_data['files'][filename]['max-size']  = int(c.return_value('file ' + filename + ' archive size'))* 1024
+      if c.exists('file ' + filename + ' archive files'):
+        config_data['files'][filename]['max-files'] = c.return_value('file ' + filename + ' archive files')
+
+  ## set system syslog console
+  if c.exists('console'):
+    config_data['console']  = {
+      '/dev/console' : {
+        'selectors' : '*.err'
+      }
+    }
+    
+  for f in c.list_nodes('console facility'):
+    if c.exists('console facility ' + f + ' level'):
+      config_data['console'] = {
+        '/dev/console' : {
+          'selectors' : generate_selectors(c, 'console facility')
+        }
+      }
+    
+  ## set system syslog host
+  if c.exists('host'):
+    rhosts = c.list_nodes('host')
+    for rhost in rhosts:
+      for fac in c.list_nodes('host ' + rhost + ' facility'):
+        if c.exists('host ' + rhost + ' facility ' + fac + ' protocol'):
+          proto = c.return_value('host ' + rhost + ' facility ' + fac + ' protocol')
+        else:
+          proto = 'udp'
+
+      config_data['hosts'].update(
+        {
+          rhost : {
+           'selectors' : generate_selectors(c, 'host ' + rhost + ' facility'), 
+            'proto' : proto
+          } 
+        }
+      )
+
+  ## set system syslog user
+  if c.exists('user'):
+    usrs = c.list_nodes('user')
+    for usr in usrs:
+      config_data['user'].update(
+        {
+          usr : {
+            'selectors' : generate_selectors(c, 'user ' + usr + ' facility')
+          }
+        }
+      )
+
+  return config_data
+
+def generate_selectors(c, config_node):
+## protocols and security are being mapped here
+## for backward compatibility with old configs
+## security and protocol mappings can be removed later
+  if c.is_tag(config_node):
+    nodes = c.list_nodes(config_node)
+    selectors = ""
+    for node in nodes:
+      lvl = c.return_value( config_node + ' ' + node + ' level')
+      if lvl == None:
+        lvl = "err"
+      if lvl == 'all':
+        lvl = '*'
+      if node == 'all' and node != nodes[-1]:
+        selectors += "*." + lvl + ";"
+      elif node == 'all':
+        selectors += "*." + lvl
+      elif node != nodes[-1]:
+        if node == 'protocols':
+          node = 'local7'
+        if node == 'security':
+          node = 'auth'
+        selectors += node + "." + lvl + ";"
+      else:
+        if node == 'protocols':
+          node = 'local7'
+        if node == 'security':
+          node = 'auth'
+        selectors += node + "." + lvl
+    return selectors
+
+def generate(c):
+  tmpl = jinja2.Template(configs, trim_blocks=True)
+  config_text = tmpl.render(c)
+  #print (config_text)
+  with open('/etc/rsyslog.d/vyos-rsyslog.conf', 'w') as f:
```

Comment:
Perhaps move it to a constant at the top of the file?

Reply:
It's only used once in that line. Still want to have a constant?